### PR TITLE
fix(api-markdown-documenter): Always display type param info when it is present

### PR DESCRIPTION
**Pull request by Josmithr**
The code as written would only display type parameter information for `class` and `interface` items. This was an oversight when making the initial implementation. The code now correctly displays type parameter docs in the `Signature` section if the API item has any.

Also updates type parameter tables to display type constraint and default excerpts, instead of an "optional" notice.

See the updated test output for an example of the changes.

### Breaking Change

These changes also update the signature of `createTypeParametersSection` to be a bit more user friendly, which is at least semantically breaking. It will now _always_ generate the section if asked. If the consumer wants to omit the section (for example when the list of type parameters is empty), they can make the call conditional on their end. The benefit is that consumers no longer have to make checks for undefined results.

### Misc

Also resolves an issue with the build pipeline for this package. Since this package uses a `file:` dev dependency on `@fluidframework/eslint-config-fluid`, we need to run the build pipeline when changes are made in that directory.

AB#7033

### CODE REVIEW

**Review by alexvy86 — APPROVED** [review]

**Comment by alexvy86** [comment]
I guess a changelog entry in the independent package to call out the semantically breaking change would make sense?

File: tools/pipelines/build-api-markdown-documenter.yml [hunk 1]
```diff
@@ -32,6 +32,7 @@ trigger:
   paths:
     include:
     - .prettierignore
+    - common/build/eslint-config-fluid # Required since api-markdown-documenter takes a `file:` dependency on this package
```
Comment (Josmithr) [inline tools/pipelines/build-api-markdown-documenter.yml:35]: This is really unrelated to the core changes being made here, but it fixes a gap in our pipeline trigger conditions.

**Review by alexvy86 — APPROVED** [review]
I don't know how common this is in our repo, especially for consumer-facing APIs, but I think this might need adjustments to handle multiple generic type parameters? Could certainly be a follow-up.

**Comment by Josmithr** [comment]
> I don't know how common this is in our repo, especially for consumer-facing APIs, but I think this might need adjustments to handle multiple generic type parameters? Could certainly be a follow-up.

Can you elaborate? The altered test cases don't illustrate it, but we do create a table entry for each in the case of multiple params. If you're referring to nested types, those should also be captured (as a single "excerpt")

E.g. https://fluidframework.com/docs/api/v2/fluid-framework/treenodeschemanonclass-interface#type-parameters

**Comment by alexvy86** [comment]
> > I don't know how common this is in our repo, especially for consumer-facing APIs, but I think this might need adjustments to handle multiple generic type parameters? Could certainly be a follow-up.
> 
> Can you elaborate? The altered test cases don't illustrate it, but we do create a table entry for each in the case of multiple params.
> 
> E.g. https://fluidframework.com/docs/api/v2/fluid-framework/treenodeschemanonclass-interface#type-parameters
> 
> If you're referring to nested types, those should also be captured (as a single "excerpt")

That linked example is exactly what I was thinking about, so it is handled already 👍 .

**Pull request merged**